packages/flutter_web_plugins/lib/src/history/utils.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:html';
+
+AnchorElement _urlParsingNode;
+
+/// Extracts the pathname part of a full [url].
+///
+/// Example: for the url `http://example.com/foo`, the extracted pathname will
+/// be `/foo`.
+String extractPathname(String url) {
+  _urlParsingNode ??= AnchorElement();
+  _urlParsingNode.href = url;
+  final String pathname = _urlParsingNode.pathname;
+  return (pathname.isEmpty || pathname[0] == '/') ? pathname : '/$pathname';
+}
+
+/// Prepends a slash to [path] if it doesn't start with a slash already.
+///
+/// If the path already starts with a slash, it'll be returned unchanged.
```

Comment:
nit: prefer present tense and non-passive form, i.e. "returns the path unchanged"